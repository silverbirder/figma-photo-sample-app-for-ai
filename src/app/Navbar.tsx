import React from "react";
import styles from "./Navbar.module.css";

const Navbar = () => {
  return (
    <div className={styles.navbar}>
      <div className={styles.logo}>photo</div>
      <div className={styles.navLinks}>
        <a href="#" className={styles.navLink}>
          Features
        </a>
        <a href="#" className={styles.navLink}>
          Pricing
        </a>
        <a href="#" className={styles.navLink}>
          Community
        </a>
        <a href="#" className={styles.navLink}>
          Support
        </a>
      </div>
      <div>
        <button className={`${styles.button} ${styles.login}`}>Log in</button>
        <button className={`${styles.button} ${styles.register}`}>
          Register
        </button>
      </div>
    </div>
  );
};

export default Navbar;
