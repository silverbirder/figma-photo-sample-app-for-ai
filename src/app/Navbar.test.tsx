import { expect, it, describe } from "vitest";
import { render } from "@testing-library/react";
import Navbar from "./Navbar";

describe("Navbar Component", () => {
  const setup = () => {
    const utils = render(<Navbar />);

    return { ...utils };
  };

  it("should render correctly and match snapshot", () => {
    // Act
    const { asFragment } = setup();

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the logo", () => {
    // Act
    const { getByText } = setup();

    // Assert
    expect(getByText("photo")).toBeDefined();
  });

  it("should render all navigation links", () => {
    // Act
    const { getByText } = setup();

    // Assert
    const navLinks = ["Features", "Pricing", "Community", "Support"];
    navLinks.forEach((linkText) => {
      expect(getByText(linkText)).toBeDefined();
    });
  });

  it("should render the login and register buttons", () => {
    // Act
    const { getByRole } = setup();

    // Assert
    expect(getByRole("button", { name: "Log in" })).toBeDefined();
    expect(getByRole("button", { name: "Register" })).toBeDefined();
  });
});
