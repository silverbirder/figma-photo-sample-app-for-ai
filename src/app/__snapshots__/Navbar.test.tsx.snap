// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar Component > should render correctly and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="_navbar_c1c3c6"
  >
    <div
      class="_logo_c1c3c6"
    >
      photo
    </div>
    <div
      class="_navLinks_c1c3c6"
    >
      <a
        class="_navLink_c1c3c6"
        href="#"
      >
        Features
      </a>
      <a
        class="_navLink_c1c3c6"
        href="#"
      >
        Pricing
      </a>
      <a
        class="_navLink_c1c3c6"
        href="#"
      >
        Community
      </a>
      <a
        class="_navLink_c1c3c6"
        href="#"
      >
        Support
      </a>
    </div>
    <div>
      <button
        class="_button_c1c3c6 _login_c1c3c6"
      >
        Log in
      </button>
      <button
        class="_button_c1c3c6 _register_c1c3c6"
      >
        Register
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navbar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="_navbar_c1c3c6"
  >
    <div
      class="_logo_c1c3c6"
    >
      photo
    </div>
    <div
      class="_navLinks_c1c3c6"
    >
      <a
        class="_navLink_c1c3c6"
        href="#"
      >
        Features
      </a>
      <a
        class="_navLink_c1c3c6"
        href="#"
      >
        Pricing
      </a>
      <a
        class="_navLink_c1c3c6"
        href="#"
      >
        Community
      </a>
      <a
        class="_navLink_c1c3c6"
        href="#"
      >
        Support
      </a>
    </div>
    <div>
      <button
        class="_button_c1c3c6 _login_c1c3c6"
      >
        Log in
      </button>
      <button
        class="_button_c1c3c6 _register_c1c3c6"
      >
        Register
      </button>
    </div>
  </div>
</DocumentFragment>
`;
